doc/stdenv/passthru.chapter.md

```diff
@@ -0,0 +1,253 @@
+# Passthru-attributes {#chap-passthru}
+[]{#var-stdenv-passthru} []{#special-variables} <!-- legacy anchors -->
+
+`passthru` is an attribute set which can be filled with arbitrary values. 
+For example:
+
+```nix
+{
+  passthru = {
+    foo = "bar";
+    baz = {
+      value1 = 4;
+      value2 = 5;
+    };
+  };
+}
+```
+
+Much of the passthru's power comes from two features:
+- You can access passthru attributes as if they were set inside a derivation (e.g. `hello.baz.value1`).
+- These values are not passed to the builder, so you can change them without triggering a rebuild.
+
+One example of the passthru's power is how it enables us to associate the package for an interpreter Python or Ruby with a set of modules built for that specific interpreter (e.g., `python.pkgs` and `ruby.gems`) and to build runtime environments with a collection of those modules (e.g., (`python.withPackages`)[#python.withpackages-function] and `ruby.withPackages`).
+<!-- python.pkgs, ruby.gems, and ruby.withPackages are mentioned but don't have dedicated sections to link -->
+
+::: {.note}
+Since `passthru` is meant for values that are useful outside the derivation (e.g. in other derivations), it has no strict schema.
+:::
+
+<!-- 
+I'm extracting this statement from the old text because I feel like it isn't very clearly-phrased and doesn't help connect with something most people know already.
+I still think we should try to make sure the pattern it models is represented in the list of common attributes, though:
+
+convey a specific dependency of your derivation which contains a program with plugins support. 
+Later, others who make derivations with plugins can use passed-through dependency to ensure that their plugin would be binary-compatible with built program. 
+-->
```

Comment:
Still undecided on this. Roughly a TODO, but I didn't want to go off searching for a real example of this and writing it up if all of this is off-base in the first place.